sentry-tracing/src/layer.rs

```diff
@@ -197,6 +198,8 @@ fn record_fields<'a, K: AsRef<str> + Into<Cow<'a, str>>>(
 pub(super) struct SentrySpanData {
     pub(super) sentry_span: TransactionOrSpan,
     parent_sentry_span: Option<TransactionOrSpan>,
+    hub: Arc<sentry_core::Hub>,
+    hub_switch_guard: Option<sentry_core::HubSwitchGuard>,
```

Comment:
I believe it would be clearer if we used a proper enum for this, as you either have a `Hub`, or a `HubSwitchGuard`.

I don’t think you are actively using the `hub` when the guard is active?

Reply:
It is used in on_exit to configure scope to switch from current span to parent's